File: 0075_sort_colors.py
# https://leetcode-cn.com/problems/sort-colors/

class Solution:
    def sortColors(self, nums):
        """
        Inputs:
            nums: list[int]

        Do not return anything, modify nums in-place instead.
        """

        n = len(nums)
        idx = 0

        # Move 0s to the front of the list.
        for i in range(n):
            if nums[i] == 0:
                nums[i] = nums[idx]
                nums[idx] = 0
                idx = idx + 1

        # Move 1s to the middle part right after 0s.
        for i in range(idx, n):
            if nums[i] == 1:
                nums[i] = nums[idx]
                nums[idx] = 1
                idx = idx + 1

File: Cpp/0075_sort_colors.cpp
// https://leetcode.cn/problems/sort-colors/

class Solution {
public:
    void sortColors(vector<int>& nums) {
        int cnt_0 = 0, cnt_1 = 0;
        for (int num: nums) {
            if (num == 0) {
                cnt_0 = cnt_0 + 1;
            }
            else if (num == 1) {
                cnt_1 = cnt_1 + 1;
            }
        }
        for (int i = 0; i < cnt_0; ++i) {
            nums[i] = 0;
        }
        for (int i = cnt_0; i < cnt_0+cnt_1; ++i) {
            nums[i] = 1;
        }
        for (int i = cnt_0+cnt_1; i < nums.size(); ++i) {
            nums[i] = 2;
        }
    }
};
